docs/decisions/0009-pylint-assertions.rst

```diff
@@ -0,0 +1,50 @@
+Pylint Assertions Standardization
+---------------------------------
+
+Status
+======
+
+Proposed
+
+Context
+=======
+
+The edx-platform test suite includes a massive number of individual assertions, written over time for different Python versions and test runners.  A nontrivial number of these currently trigger deprecation warnings, as they are slated for replacement or removal in future Python versions.  A very high percentage of them use assertion methods from Python's ``unittest`` module, which were inspired by Java's jUnit library long ago and have notable drawbacks vs. the `assert statement rewriting`_ built into our current test runner, pytest.  In particular:
+
+* ``assert`` statements are typically shorter and easier to read.  For example:
+
+  * ``self.assertNotIn("session_key", self.client.session)`` (unittest) becomes
+
+  * ``assert "session_key" not in self.client.session`` (pytest)
+
+* The output of failed pytest-rewritten assertions is often more helpful than their ``unittest`` equivalents.
+
+* Using ``assert`` eliminates the need to remember which of 20+ methods is most appropriate for your current assertion, or to look up the correct order of arguments for the one you choose.
```

Comment:
At least, these assertions are documented, right? https://docs.python.org/3/library/unittest.html
This is not the case for the pytest library. For instance, would you be able to predict the console output of the following test?

```
def test_sum():
    assert 3 == 1 + 1
```

You can't, based on the documenation. It's actually:

```
FAILED test_pouac.py::test_pouac - assert 3 == (1 + 1)
```

Which means that pytest does not print the actual value of the right-hand side of the equation. To get the actual value, we need to run this in verbose mode:

```
$ pytest -v test_sum.py
    def test_sum():                                                                                                                                                                                              
>       assert 3 == 1 + 1                                                                                
E       assert 3 == 2                                                                                                                                                                                              
E         +3                                                                                             
E         -2                                                                                             
                                                                                                                                                                                                                   
test_sum.py:2: AssertionError 
```

Can we talk about the console output here? Even after years of usage I still do not know how to interpret the following:

```
E         +3                                                                                             
E         -2                                                                                             
```